### PR TITLE
Backend cleanup

### PR DESCRIPTION
**Pull request by Spreeuw**
This PR focuses primarily on escaping / sanitizing settings windows and a few other backend contexts.

Please review code (checking to see if any strings were missing) as well as output (verifying that the changes made still result in the correct output).

### CODE REVIEW

**Comment by alexmigf** [comment]
What you think of applying some escaping here?

https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/2f8f207e201c7cc2c1c46fb967ddde44a6dec2b1/includes/documents/abstract-wcpdf-order-document.php#L423

https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/2f8f207e201c7cc2c1c46fb967ddde44a6dec2b1/includes/documents/abstract-wcpdf-order-document.php#L429

https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/2f8f207e201c7cc2c1c46fb967ddde44a6dec2b1/includes/documents/abstract-wcpdf-order-document.php#L905

https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/2f8f207e201c7cc2c1c46fb967ddde44a6dec2b1/includes/documents/abstract-wcpdf-order-document.php#L908

File: includes/views/wcpdf-extensions.php [hunk 1]
```diff
@@ -11,93 +12,93 @@
...
 			</li>
 		<?php } ?>
 
 		<?php
-		if (!class_exists('WooCommerce_Ext_PrintOrders')) {
+		if ( ! class_exists( 'WooCommerce_Ext_PrintOrders' ) ) {
 			?>
 			<li>
-				<?php _e('Automatically send new orders or packing slips to your printer, as soon as the customer orders!', 'woocommerce-pdf-invoices-packing-slips' )?>
+				<?php esc_html_e( 'Automatically send new orders or packing slips to your printer, as soon as the customer orders!', 'woocommerce-pdf-invoices-packing-slips' )?>
```
Comment (alexmigf) [inline includes/views/wcpdf-extensions.php:94]: Missing ending semicolon

File: includes/views/wcpdf-extensions.php [hunk 2]
```diff
@@ -11,93 +12,93 @@
...
+	<?php $no_pro = ! class_exists( 'WooCommerce_PDF_IPS_Pro' ) && ! class_exists( 'WPO_WCPDF_Templates' ); ?>
+	<img src="<?php echo esc_url( WPO_WCPDF()->plugin_url() . '/assets/images/wpo-helper.png' ); ?>" class="wpo-helper">
+	<h3><?php esc_html_e( 'Check out these premium extensions!', 'woocommerce-pdf-invoices-packing-slips' ); ?></h3>
+	<i>(<?php esc_html_e( 'click items to read more', 'woocommerce-pdf-invoices-packing-slips' ); ?>)</i>
 	<ul class="extensions">
 		<?php if ( $no_pro ): ?>
 			<!-- No Pro extensions: Ad for PDF bundle -->
 			<li>
-				<?php _e('Premium PDF Invoice bundle: Everything you need for a perfect invoicing system', 'woocommerce-pdf-invoices-packing-slips' )?>
+				<?php esc_html_e( 'Premium PDF Invoice bundle: Everything you need for a perfect invoicing system', 'woocommerce-pdf-invoices-packing-slips' )?>
```
Comment (alexmigf) [inline includes/views/wcpdf-extensions.php:23]: Missing ending semicolon

File: includes/views/wcpdf-extensions.php [hunk 3]
```diff
@@ -11,93 +12,93 @@
...
+					<li><?php echo wp_kses_post( __( 'Attach <b>up to 3 static files</b> (for example a terms & conditions document) to the WooCommerce emails of your choice.', 'woocommerce-pdf-invoices-packing-slips' ) ); ?></li>
+					<li><?php echo wp_kses_post( __( 'Use <b>separate numbering systems</b> and/or format for proforma invoices and credit notes or utilize the main invoice numbering system', 'woocommerce-pdf-invoices-packing-slips' ) ); ?></li>
+					<li><?php echo wp_kses_post( __( '<b>Customize</b> the <b>shipping & billing address</b> format to include additional custom fields, font sizes etc. without the need to create a custom template.', 'woocommerce-pdf-invoices-packing-slips' ) ); ?></li>
+					<li><?php echo wp_kses_post( __( 'Use the plugin in multilingual <b>WPML</b> setups', 'woocommerce-pdf-invoices-packing-slips' ) ); ?></li>
 				</ul>
-				<?php _e('Advanced, customizable templates', 'woocommerce-pdf-invoices-packing-slips' )?>
+				<?php esc_html_e( 'Advanced, customizable templates', 'woocommerce-pdf-invoices-packing-slips' )?>
```
Comment (alexmigf) [inline includes/views/wcpdf-extensions.php:35]: Missing ending semicolon

File: includes/views/wcpdf-extensions.php [hunk 4]
```diff
@@ -11,93 +12,93 @@
...
 				<ul>
-					<li><?php _e( 'Completely customize the invoice contents (prices, taxes, thumbnails) to your needs with a drag & drop customizer', 'woocommerce-pdf-invoices-packing-slips' ); ?></li>
-					<li><?php _e( 'Two extra stylish premade templates (Modern & Business)', 'woocommerce-pdf-invoices-packing-slips' ); ?></li>
+					<li><?php echo wp_kses_post( __( 'Completely customize the invoice contents (prices, taxes, thumbnails) to your needs with a drag & drop customizer', 'woocommerce-pdf-invoices-packing-slips' ) ); ?></li>
+					<li><?php echo wp_kses_post( __( 'Two extra stylish premade templates (Modern & Business)', 'woocommerce-pdf-invoices-packing-slips' ) ); ?></li>
 				</ul>
-				<?php _e('Upload automatically to dropbox', 'woocommerce-pdf-invoices-packing-slips' )?>
+				<?php esc_html_e( 'Upload automatically to dropbox', 'woocommerce-pdf-invoices-packing-slips' )?>
```
Comment (alexmigf) [inline includes/views/wcpdf-extensions.php:40]: Missing ending semicolon

File: includes/views/wcpdf-extensions.php [hunk 5]
```diff
@@ -11,93 +12,93 @@
...
 			</li>
 		<?php endif; ?>
 		<?php
 		// NO BUNDLE: separate ads
-		if (!class_exists('WooCommerce_PDF_IPS_Pro') && !$no_pro) {
+		if ( ! class_exists( 'WooCommerce_PDF_IPS_Pro' ) && ! $no_pro ) {
 			?>
 			<li>
-				<?php _e('Go Pro: Proforma invoices, credit notes (=refunds) & more!', 'woocommerce-pdf-invoices-packing-slips' )?>
+				<?php esc_html_e( 'Go Pro: Proforma invoices, credit notes (=refunds) & more!', 'woocommerce-pdf-invoices-packing-slips' )?>
```
Comment (alexmigf) [inline includes/views/wcpdf-extensions.php:54]: Missing ending semicolon

File: includes/views/wcpdf-extensions.php [hunk 6]
```diff
@@ -11,93 +12,93 @@
...
 			</li>
 		<?php } ?>
 
 		<?php
-		if (!class_exists('WPO_WC_Smart_Reminder_Emails')) {
+		if ( ! class_exists( 'WPO_WC_Smart_Reminder_Emails' ) ) {
 			?>
 			<li>
-				<?php _e('Automatically send payment reminders to your customers', 'woocommerce-pdf-invoices-packing-slips' )?>
+				<?php esc_html_e( 'Automatically send payment reminders to your customers', 'woocommerce-pdf-invoices-packing-slips' )?>
```
Comment (alexmigf) [inline includes/views/wcpdf-extensions.php:74]: Missing ending semicolon

File: includes/views/wcpdf-extensions.php [hunk 7]
```diff
@@ -11,93 +12,93 @@
...
 				<div class="more" style="display:none;">
-				<?php _e('WooCommerce Smart Reminder emails', 'woocommerce-pdf-invoices-packing-slips' )?>
+				<?php esc_html_e( 'WooCommerce Smart Reminder emails', 'woocommerce-pdf-invoices-packing-slips' )?>
```
Comment (alexmigf) [inline includes/views/wcpdf-extensions.php:76]: Missing ending semicolon

File: includes/views/wcpdf-extensions.php [hunk 8]
```diff
@@ -106,29 +107,29 @@
...
 
 		<?php
-		if (!class_exists('WooCommerce_PDF_IPS_Templates') && !class_exists('WPO_WCPDF_Templates') && !$no_pro) {
+		if ( ! class_exists( 'WooCommerce_PDF_IPS_Templates' ) && ! class_exists( 'WPO_WCPDF_Templates' ) && ! $no_pro ) {
 			$template_link = '<a href="https://wpovernight.com/downloads/woocommerce-pdf-invoices-packing-slips-premium-templates/" target="_blank">wpovernight.com</a>';
 			$email_link = '<a href="mailto:support@wpovernight.com">support@wpovernight.com</a>'
 			?>
 			<li>
-				<?php _e('Advanced, customizable templates', 'woocommerce-pdf-invoices-packing-slips' )?>
+				<?php esc_html_e( 'Advanced, customizable templates', 'woocommerce-pdf-invoices-packing-slips' )?>
```
Comment (alexmigf) [inline includes/views/wcpdf-extensions.php:115]: Missing ending semicolon

**Comment by alexmigf** [comment]
And this?

https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/2f8f207e201c7cc2c1c46fb967ddde44a6dec2b1/includes/class-wcpdf-admin.php#L92-L98

https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/2f8f207e201c7cc2c1c46fb967ddde44a6dec2b1/includes/class-wcpdf-admin.php#L132-L133

https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/2f8f207e201c7cc2c1c46fb967ddde44a6dec2b1/includes/class-wcpdf-admin.php#L333

https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/2f8f207e201c7cc2c1c46fb967ddde44a6dec2b1/includes/class-wcpdf-admin.php#L620

https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/2f8f207e201c7cc2c1c46fb967ddde44a6dec2b1/includes/class-wcpdf-admin.php#L669

https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/2f8f207e201c7cc2c1c46fb967ddde44a6dec2b1/includes/class-wcpdf-admin.php#L754-L775

https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/2f8f207e201c7cc2c1c46fb967ddde44a6dec2b1/includes/class-wcpdf-admin.php#L869

https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/2f8f207e201c7cc2c1c46fb967ddde44a6dec2b1/includes/class-wcpdf-admin.php#L904-L907

File: includes/class-wcpdf-settings-callbacks.php [hunk 9]
```diff
@@ -208,30 +206,30 @@ public function select( $args ) {
 				$multiple = '';
 			}
 
-			$placeholder = isset($placeholder) ? esc_attr( $placeholder ) : '';
-			$title = isset($title) ? esc_attr( $title ) : '';
+			$placeholder = isset( $placeholder ) ? esc_attr( $placeholder ) : '';
+			$title = isset( $title ) ? esc_attr( $title ) : '';
```
Comment (alexmigf) [inline includes/class-wcpdf-settings-callbacks.php:210]: Maybe use `! empty` like you're doing for the `$description`?

File: includes/class-wcpdf-settings-callbacks.php [hunk 10]
```diff
@@ -208,30 +206,30 @@ public function select( $args ) {
...
 			$css = 'width:400px';
-			printf( '<select id="%1$s" name="%2$s" data-placeholder="%3$s" title="%4$s" class="%5$s" style="%6$s" %7$s>', $id, $setting_name, $placeholder, $title, $class, $css, $multiple );
+			printf( '<select id="%1$s" name="%2$s" data-placeholder="%3$s" title="%4$s" class="%5$s" style="%6$s" %7$s>', esc_attr( $id ), esc_attr( $setting_name ), esc_attr( $placeholder ), esc_attr( $title ), esc_attr( $class ), esc_attr( $css ), $multiple );
 		} else {
-			printf( '<select id="%1$s" name="%2$s">', $id, $setting_name );
+			printf( '<select id="%1$s" name="%2$s">', esc_attr( $id ), esc_attr( $setting_name ) );
 		}
 
 		foreach ( $options as $key => $label ) {
 			if ( isset( $multiple ) && is_array( $current ) ) {
```
Comment (alexmigf) [inline includes/class-wcpdf-settings-callbacks.php:219]: `! empty()`?

File: includes/class-wcpdf-settings-callbacks.php [hunk 11]
```diff
@@ -290,38 +288,38 @@ public function radio_button( $args ) {
...
-			echo "<p><strong>{$header}</strong>:</p>";
+		if ( ! empty( $header ) ) {
+			echo wp_kses_post( "<p><strong>{$header}</strong>:</p>" );
 		}
 
-		printf('<p class="%s multiple-text-input">', $id);
+		printf('<p class="%s multiple-text-input">', esc_attr( $id ) );
 		foreach ($fields as $name => $field) {
 			$size = $field['size'];
 			$placeholder = isset( $field['placeholder'] ) ? $field['placeholder'] : '';
```
Comment (alexmigf) [inline includes/class-wcpdf-settings-callbacks.php:298]: `! empty()`?

File: includes/class-wcpdf-settings-callbacks.php [hunk 12]
```diff
@@ -290,38 +288,38 @@ public function radio_button( $args ) {
...
 
-			if (isset($field['label_width'])) {
+			if ( isset( $field['label_width'] ) ) {
```
Comment (alexmigf) [inline includes/class-wcpdf-settings-callbacks.php:300]: `! empty()`?

File: includes/class-wcpdf-settings-callbacks.php [hunk 13]
```diff
@@ -361,51 +357,51 @@ public function multiple_checkboxes( $args ) {
...
-				$in_height = floatval($header_logo_height);
+				$in_height = floatval( $header_logo_height );
 			} else {
 				// don't display resolution
 			}
 
-			printf('<img src="%1$s" style="display:block" id="img-%4$s"/>', $attachment_src, $attachment_width, $attachment_height, $id );
-			if ( !empty($attachment_height) && !empty($in_height) ) {
-				$attachment_resolution = round(absint($attachment_height)/$in_height);
+			printf( '<img src="%1$s" style="display:block" id="img-%4$s"/>', $attachment_src, $attachment_width, $attachment_height, $id );
```
Comment (alexmigf) [inline includes/class-wcpdf-settings-callbacks.php:377]: Escape variables here?

File: includes/class-wcpdf-settings-callbacks.php (hunk 12 above)
Comment (alexmigf) [reply includes/class-wcpdf-settings-callbacks.php:300]: Are there many others, maybe do a search in VS Code if you want to change them.

**Comment by alexmigf** [comment]
Maybe this shouldn't be translated like the other log messages?

https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/277d37eac7eaaf34be28fe957fa3f0fe1dc25a79/includes/class-wcpdf-settings.php#L399

File: includes/class-wcpdf-settings-callbacks.php (hunk 11 above)
Comment (Spreeuw) [reply includes/class-wcpdf-settings-callbacks.php:298]: `$field` exists so `isset()` is safe here

File: includes/class-wcpdf-settings-callbacks.php (hunk 12 above)
Comment (Spreeuw) [reply includes/class-wcpdf-settings-callbacks.php:300]: :+1:

**Comment by alexmigf** [comment]
Indentation:

https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/ff98e36eb46cea9f6cb5edc8f3f4d7644e1a01de/includes/documents/abstract-wcpdf-order-document.php#L228-L230

**Comment by alexmigf** [comment]
Escape variables:

https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/ff98e36eb46cea9f6cb5edc8f3f4d7644e1a01de/includes/documents/abstract-wcpdf-order-document.php#L615

**Comment by alexmigf** [comment]
Escape here?

https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/ff98e36eb46cea9f6cb5edc8f3f4d7644e1a01de/includes/class-wcpdf-admin.php#L399-L407

**Comment by alexmigf** [comment]
This too?

https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/ff98e36eb46cea9f6cb5edc8f3f4d7644e1a01de/includes/class-wcpdf-main.php#L830

**Comment by Spreeuw** [comment]
> Escape here? Since this is displayed in a metabox.
> 
> https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/ff98e36eb46cea9f6cb5edc8f3f4d7644e1a01de/includes/class-wcpdf-admin.php#L399-L407

No, that's too early. It's escaped in the output function already.



> This too?
> 
> https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/ff98e36eb46cea9f6cb5edc8f3f4d7644e1a01de/includes/class-wcpdf-main.php#L830

Same here, this is only the data, it shouldn't be escaped until it's printed

**Review by YordanSoares — APPROVED** [review]
✔️ Looks good to me!

**Pull request merged**